thorlcr/master/thdemonserver.cpp

```diff
@@ -227,6 +251,22 @@ class DeMonServer : public CSimpleInterface, implements IDeMonServer
     void startGraph(CGraphBase *graph)
     {
         synchronized block(mutex);
+
+        IConstWorkUnit & wu =  graph->queryJob().queryWorkUnit();
+        workunitCost = aggregateCost(&wu);
+
+        const IPropertyTree *costs = queryCostsConfiguration();
+        double softLimit = 0.0, hardLimit = 0.0;
+        if (costs)
+        {
+            softLimit = costs->getPropReal("@limit");
+            hardLimit = costs->getPropReal("@hardlimit");
+        }
+        double tmpcostLimit = wu.getDebugValueReal("maxCost", softLimit);
+        if (hardLimit && (tmpcostLimit > hardLimit))
```

Comment:
Currently hard limits can be avoided by setting maxcost to 0..  Should be something like:

`hardLimit && ((tmpcostLimit == 0) || (tmpcostLimit > hardLimit))
`